### DISCUSSION TITLE
Low Volume

### DISCUSSION THREAD

**Discussion by unknown0816** [CarlosDerSeher/snapclient#66] [Q&A]
I flashed the snapclient to a AI Thinker ESP32 A1S Audio Kit. I just had to change some PINs and the Audio Kit is playing audio from the snapserver. 
However, the volume is low. Further, if I try to increase or decrease volume with a snapcast control the volume control does not do anything, but I can see that the information is processed in the monitoring:
I (39651) SC: Buffer length:  500
I (39652) SC: Latency:        0
I (39652) SC: Mute:           0
I (39654) SC: Setting volume: 100

The Mute button in the snapcast control works quiet well. 

I already tried to change some configuration like the DSP processor, but nothing is working.

Does somebody has an idea?

**Comment by unknown0816** [comment]
Okay. I figured out a solution. [Here](https://github.com/Ai-Thinker-Open/ESP32-A1S-AudioKit/issues/36)  is described which registers/ DAC CONTROLS are used for the volume. After that I had to change the registers/ DAC CONTROLS in the set_volume function of the es8388 (es8388_set_voice_volume function in es8388.c).

As far as I could understand, in the standard config the volume is only set for the output1 but the jack output is output2 (this was just set to 0 - -45dB). That's why it was hear-able but not that loud and why changing the volume via snapcast did not have any effect.  After modifying the volume set to output2, it was brutally loud, because now the lowest volume value was the -45dB. To change this I used another 2 registers that are as far as I understand more globally and have a lower dB range (up to -96dB). With this change it works quiet well.
However, it would be possible to look even deeper into that and find an even better solution. [This](https://github.com/Ai-Thinker-Open/ESP32-A1S-AudioKit/issues/42) one here looked promising, too.

**Reply by CarlosDerSeher** [reply]
If possible you could set both outputs to provide good volume and then we can merge your pull request?

**Reply by unknown0816** [reply]
This would be possible. I guess I should add this as some kind of new board to not mess up the existing boards. Let's see if I can figure out how this can be done.

**Reply by unknown0816** [reply]
All done. Pull request is open.

**Reply by CarlosDerSeher** [reply]
You can also add a driver using custom board component
